TrackletAlgorithm/VMRouterCM.h

```diff
@@ -0,0 +1,434 @@
+		const int vmbits, const int finebits) {
+
+	auto finebin = (phi.range(phi.length() - vmbits - 1, phi.length() - vmbits - finebits));
+
+	return finebin;
+}
+
+// Get the corrected phi, i.e. phi at the average radius of the barrel
+// Corrected phi is used by ME and TE memories in the barrel
+template<regionType InType>
+inline typename AllStub<InType>::ASPHI getPhiCorr(
+		const typename AllStub<InType>::ASPHI phi,
+		const typename AllStub<InType>::ASR r,
+		const typename AllStub<InType>::ASBEND bend, const int phiCorrTable[]) {
+
+	if (InType == DISKPS || InType == DISK2S)
+		return phi; // Do nothing if disks
+
+	constexpr auto nrbins = 1 << nbitsrphicorrtable; // The number of bins for r
+
+	ap_uint<nbitsrphicorrtable> rbin = (r + (1 << (r.length() - 1)))
+			>> (r.length() - nbitsrphicorrtable); // Which bin r belongs to. Note r = 0 is mid radius
+	auto index = bend * nrbins + rbin; // Index for where we find our correction value
+	auto corrValue = phiCorrTable[index]; // The amount we need to correct our phi
+
+	auto phicorr = phi - corrValue; // the corrected phi
+
+	// Check for overflow
+	if (phicorr < 0)
+		phicorr = 0; // can't be less than 0
+	if (phicorr >= 1 << phi.length())
+		phicorr = (1 << phi.length()) - 1;  // can't be more than the max value
+
+	return phicorr;
+}
+
+
+// Returns a ME/TE stub with all the values set
+template<class T, regionType InType, regionType OutType, int Layer, int Disk, bool isMEStub>
+inline T createVMStub(const InputStub<InType> inputStub,
+		const int index, const bool negDisk, const int lutTable[],
+		const int phiCorrTable[], int& slot) {
+
+	// The stub that is going to be returned
+	T stub;
+
+	// Values from InputStub
+	auto z = inputStub.getZ();
+	auto r = inputStub.getR();
+	auto bend = inputStub.getBend();
+	auto phi = inputStub.getPhi();
+	auto phicorr = getPhiCorr<InType>(phi, r, bend, phiCorrTable); // Corrected phi, i.e. phi at nominal radius
+
+	int nbitsr = r.length(); // Number of bits for r
+	int nbitsz = z.length(); // Number of bits for z
+	int nbitsfinerz = stub.getFineZ().length(); // Number of bits for finer/z
+	int nbitsfinephi = stub.getFinePhi().length();  // Number of bits for finephi
+
+	// Number of bits for table indices
+	constexpr int nbitsztable =
+			(Layer) ? nbitsztablelayer : nbitsztabledisk; // Number of MSBs of z used in LUT table
+	constexpr int nbitsrtable =
+			(Layer) ? nbitsrtablelayer : nbitsrtabledisk; // Number of MSBs of r used in LUT table
+	constexpr auto vmbits =
+			(Layer) ? nbitsvmlayer[Layer - 1] : nbitsvmdisk[Disk - 1]; // Number of bits for standard VMs
+	constexpr unsigned int nbitsall = (Layer) ? nbitsallstubs[Layer-1] : nbitsallstubs[N_LAYER+Disk-1]; // Number of bits for the number of Alltub memories in a layer/disk
+
+	// Set values to VMStub
+	stub.setBend(bend);
+	stub.setIndex(index);
+	stub.setFinePhi(
+				iphivmFineBins<InType>(phicorr, nbitsall + vmbits, nbitsfinephi));
```

Comment:
Could you please go through this file and make sure the indentation is consistent and reasonable? For example, I'm not sure why this is on a separate line and indented so much.